pkg/puller/puller.go

```diff
@@ -406,13 +304,19 @@ func (p *Puller) liveSyncWorker(ctx context.Context, peer swarm.Address, bin uin
 			return
 		default:
 		}
+
 		top, ruid, err := p.syncer.SyncInterval(ctx, peer, bin, from, pullsync.MaxCursor)
 		if err != nil {
-			loggerV2.Debug("liveSyncWorker exit on sync error", "peer_address", peer, "bin", bin, "from", from, "error", err)
-			if ruid == 0 {
-				p.metrics.LiveWorkerErrCounter.Inc()
+			p.metrics.LiveWorkerErrCounter.Inc()
+
+			if errors.Is(err, context.Canceled) {
```

Comment:
This is risky to do if the peer is not reachable. We will stay in this retry loop which could cause a lot of CPU usage. Maybe we should have some retry counter.

Also, I would add a log message to say which attempt it is when we call SyncInterval again. This can help us identify this problem.

Also unit test needs to be added for this behaviour.